server/src/main/java/org/opensearch/cluster/routing/WeightedRoundRobin.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Weighted Round Robin Scheduling policy
+ *
+ */
+public class WeightedRoundRobin<T> {
+
+    private List<WeightedRoundRobin.Entity<T>> entities;
+
+    public WeightedRoundRobin(List<WeightedRoundRobin.Entity<T>> entities) {
+        this.entities = entities;
+    }
+
+    /**
+     * *
+     * @return list of entities that is ordered using weighted round-robin scheduling
+     * http://kb.linuxvirtualserver.org/wiki/Weighted_Round-Robin_Scheduling
+     */
+    public List<WeightedRoundRobin.Entity<T>> orderEntities() {
```

Comment:
do we need this logic or can we simplify we below ? 

Lets say the weights are `4:3:1` for `a:b:c` . Can we just just return `a a a a b b b c`  ? This should be good as we are shuffling it later on .

Reply:
We are using `RotationShardShuffler` which rotates the output by seed (1). So I think if we just return `a a a a b b b c`, it will send all the request to node with highest weight and so on. With existing algorithm, the requests will be little interleaved `a, a, b, a, b, a, b, c`

Reply:
I agree, we are initializing ```RotationShardShuffler``` using a random number. But on every request for same shard, we are incrementing the seed by 1. ```int seed = shuffler.nextSeed();```. Let me know if I am missing anything.